extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/security/MTLS.java

```diff
@@ -0,0 +1,298 @@
+import io.quarkus.vertx.http.runtime.security.MtlsAuthenticationMechanism;
+import io.smallrye.common.annotation.Experimental;
+import io.smallrye.config.SmallRyeConfigBuilder;
+import io.vertx.core.http.ClientAuth;
+
+/**
+ * This class provides a way to create the mutual TLS client authentication mechanism. The {@link HttpAuthenticationMechanism}
+ * created with this class can be registered using the {@link HttpSecurity#mechanism(HttpAuthenticationMechanism)} method.
+ */
+@Experimental("This API is currently experimental and might get changed")
+public interface MTLS {
+
+    /**
+     * Creates the mutual TLS client authentication mechanism.
+     * The client authentication is accepted if presented by a client.
+     * Use this method if the client authentication is only required for certain routes
+     * and secure these routes with HTTP permissions or standard security annotations.
+     *
+     * @param tlsConfigurationName the name of the configuration, cannot be {@code null}, cannot be {@code <default>}
+     * @param tlsConfiguration the configuration cannot be {@code null}
+     * @return MtlsAuthenticationMechanism
+     * @see VertxHttpBuildTimeConfig#tlsClientAuth() for more information about {@link ClientAuth#REQUEST}
+     * @see Builder#tls(String, TlsConfiguration) for information about method parameters and TLS config registration
+     */
+    static MtlsAuthenticationMechanism request(String tlsConfigurationName, TlsConfiguration tlsConfiguration) {
+        return builder().authentication(ClientAuth.REQUEST).tls(tlsConfigurationName, tlsConfiguration).build();
+    }
+
+    /**
+     * Creates the mutual TLS client authentication mechanism.
+     * The client authentication is accepted if presented by a client.
+     * Use this method if the client authentication is only required for certain routes
+     * and secure these routes with HTTP permissions or standard security annotations.
+     *
+     * @return MtlsAuthenticationMechanism
+     * @see VertxHttpBuildTimeConfig#tlsClientAuth() for more information
+     */
+    static MtlsAuthenticationMechanism request() {
+        return builder().authentication(ClientAuth.REQUEST).build();
+    }
+
+    /**
+     * Creates the mutual TLS client authentication mechanism. This mechanism always require the client authentication.
+     *
+     * @param tlsConfigurationName the name of the configuration, cannot be {@code null}, cannot be {@code <default>}
+     * @param tlsConfiguration the configuration cannot be {@code null}
+     * @return MtlsAuthenticationMechanism
+     * @see VertxHttpBuildTimeConfig#tlsClientAuth() for more information about {@link ClientAuth#REQUIRED}
+     * @see Builder#tls(String, TlsConfiguration) for information about method parameters and TLS config registration
+     */
+    static MtlsAuthenticationMechanism required(String tlsConfigurationName, TlsConfiguration tlsConfiguration) {
```

Comment:
@michalvavrik IMHO we still need to keep `required(String tlsConfigurationName)` (and for `request`) only as users still can do TLS config registration independently as was shown before in any case

Reply:
> @michalvavrik IMHO we still need to keep `required(String tlsConfigurationName)` (and for `request`) only as users still can do TLS config registration independently as was shown before in any case

done